api/maven-api-core/src/main/java/org/apache/maven/api/services/Source.java

```diff
@@ -20,17 +20,35 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.file.Path;
 
 import org.apache.maven.api.annotations.Experimental;
+import org.apache.maven.api.annotations.Nonnull;
+import org.apache.maven.api.annotations.Nullable;
 
 /**
- * The source for a project's XML model.
+ * The source for parsing an object.
  *
  * @since 4.0.0
  */
 @Experimental
 public interface Source {
+
+    /**
+     * Provides access the file to be parsed, if this source is backed by a file.
```

Comment:
access to

or perhaps just "Returns the path of the file being parsed"

Reply:
No, because the Source is not limited to a file.